### PR TITLE
Adds IntElementSize

### PR DESCRIPTION
**Pull request by dlurton**
Implements #23 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/com/amazon/ionelement/api/IonElement.kt [hunk 1]
```diff
@@ -110,10 +109,34 @@ interface TimestampElement : IonElement {
...
+     *
+     * @throws IonElementConstraintException if [integerSize] is [IntElementSize.BIG_INTEGER].
+     */
     val longValue: Long
+
+    /**
+     * This property may always be used to access the integer value of this [IntElement] regardless of [integerSize].
+     *
+     * However, if it is expected that all values will fall between [Long.MIN_VALUE] and [Long.MAX_VALUE], then
+     * [longValue] should be used instead for maximum performance.
```
Comment (therapon) [inline src/com/amazon/ionelement/api/IonElement.kt:138]: Are we trying to say that for perf if it fits in a Long use longValue else use this bigIntegerValue? I would just say that in one sentence.

File: src/com/amazon/ionelement/api/IonElementException.kt [hunk 2]
```diff
@@ -15,16 +15,25 @@
...
+    description: String,
+    cause: Throwable? = null
+) : IonElementException(location, description, cause)
+
+/** Exception thrown by [IonElement] accessor functions to indicate a type or nullness constraint has been violated. */
+class IonElementConstraintException(
+    location: IonLocation?,
+    description: String,
+    cause: Throwable? = null
+) : IonElementException(location, description, cause)
```
Comment (therapon) [inline src/com/amazon/ionelement/api/IonElementException.kt:39]: I just review this in another PR, I think. Did your changes get crossed here?

File: src/com/amazon/ionelement/api/IonElementExtensions.kt [hunk 3]
```diff
@@ -63,30 +63,30 @@ inline fun <reified T: IonElement> T.withoutMetas(): T =
 /**
```
Comment (therapon) [inline src/com/amazon/ionelement/api/IonElementExtensions.kt:63]: Changes in this file look identical to one of the other PRs

File: src/com/amazon/ionelement/api/IonElementException.kt (hunk 2 above)
Comment (dlurton) [reply src/com/amazon/ionelement/api/IonElementException.kt:39]: oh yeah, because I based this off of another PR.  will push that one and rebase first.

**Comment by dlurton** [comment]
Since this builds on #27, I will wait for it to be merged before revising this PR.

**Comment by dlurton** [comment]
Removed the changes from the other PR and applied feedback.

**Pull request merged**